### PR TITLE
refactor(sync): refactoring syncing process

### PR DESCRIPTION
**Pull request by b00f**
## Description

This PR includes these changes:

1- `HeartBeat` message is removed.
2- `HelloAck` message is added. Once a node receives a `HelloAck`, it can start sending messages.
3- `MaxOpenSessions` is removed and the maximum number of sessions should be the same as the number of connected peers.
4- Syncing process is based on time, not height. If the last block time is more than 20 seconds behind the current time, the node starts syncing.

### CODE REVIEW

**Review by kehiy — APPROVED** [review]
LGTM.
That's exactly what we discussed!

File: sync/bundle/message/hello.go [hunk 1]
```diff
@@ -7,30 +7,25 @@ import (
...
-	BlockHash   hash.Hash        `cbor:"10,keyasint"`
+	PeerID      peer.ID           `cbor:"1,keyasint"`
+	Agent       string            `cbor:"2,keyasint"`
+	Moniker     string            `cbor:"3,keyasint"`
+	PublicKeys  []*bls.PublicKey  `cbor:"4,keyasint"`
+	Signature   *bls.Signature    `cbor:"5,keyasint"`
+	Height      uint32            `cbor:"6,keyasint"`
+	Services    services.Services `cbor:"7,keyasint"`
+	GenesisHash hash.Hash         `cbor:"8,keyasint"`
+	BlockHash   hash.Hash         `cbor:"10,keyasint"`
```
Comment (amirvalhalla) [inline sync/bundle/message/hello.go:24]: `cbor:"9"` has missed, please fix that.

**Review by amirvalhalla — CHANGES_REQUESTED** [review]
@b00f 
please resolve request changes.

**Review by amirvalhalla — APPROVED** [review]

**Comment by b00f** [comment]
@amirvalhalla could you please double check it. If anything missed, mention me.

File: sync/bundle/message/hello.go (hunk 1 above)
Comment (amirvalhalla) [reply sync/bundle/message/hello.go]: @b00f 
cbor:"9" has missed, please fix it.

**Review by amirvalhalla — CHANGES_REQUESTED** [review]
@b00f 
please resolve request changes.

File: sync/bundle/message/hello.go [hunk 2]
```diff
@@ -7,30 +7,25 @@ import (
...
-	BlockHash   hash.Hash        `cbor:"10,keyasint"`
+	PeerID      peer.ID           `cbor:"1,keyasint"`
+	Agent       string            `cbor:"2,keyasint"`
+	Moniker     string            `cbor:"3,keyasint"`
+	PublicKeys  []*bls.PublicKey  `cbor:"4,keyasint"`
+	Signature   *bls.Signature    `cbor:"5,keyasint"`
+	Height      uint32            `cbor:"6,keyasint"`
+	Services    services.Services `cbor:"7,keyasint"`
+	GenesisHash hash.Hash         `cbor:"8,keyasint"`
+	BlockHash   hash.Hash         `cbor:"9,keyasint"`
```
Comment (themantre) [inline sync/bundle/message/hello.go:24]: @amirvalhalla it is fixed here. Please check again.

Comment (b00f) [inline sync/bundle/message/hello.go:24]: @amirvalhalla please check here

**Pull request merged**